lib/get-config.js

```diff
@@ -31,23 +31,39 @@ function resolveConfigPath(configPath, pluginPath) {
 
 }
 
-function ensureRootProperties(config) {
-  config.rules = config.rules || {};
-  config.pending = config.pending || [];
-  config.ignore = config.ignore || [];
-  config.plugins = config.plugins || [];
+function ensureRootProperties(config, source, options) {
+  let logger = options.console || console;
+
+  config.rules = assign({}, source.rules || {});
+  config.pending = (source.pending || []).slice();
+  config.ignore = (source.ignore || []).slice();
+
+  let plugins = source.plugins || [];
+  config.plugins = Array.isArray(plugins) ? plugins.slice() : assign({}, plugins);
+
+
+  let extendedList = [];
+  if (source.extends) {
+    if (typeof source.extends === 'string') {
+      extendedList = [source.extends];
+    } else if (source.extends instanceof Array) {
```

Comment:
Should use `Array.isArray` here instead of `instanceof`.

Reply:
fixed